apps/core/scenes/results/ResultsListItem.js

```diff
@@ -1,61 +1,249 @@
-   *
-   * NOTE: ugly form of string creation below is used for now because
-   * using ?? inside template literals caused problems with code styling
-   * inside VS Code (this part will be probably completely replaced
-   * in next iteration, so it's temporary solution)
-   */
-  const itemString =
-    (flyFrom ?? 'Unknown destination') +
-    ' / ' +
-    (localDeparture ?? 'Unknown departure') +
-    ' -> ' +
-    (flyTo ?? 'Unknown destination') +
-    ' / ' +
-    (localArrival ?? 'Unknown arrival') +
-    ' | ' +
-    (price ?? 'Unknown price');
-  return <Text style={styles.listItem}>{itemString}</Text>;
+// @TODO make TripSectorProps and CarrierData exportable from universal-components
+export type CarrierData = {
+  code: string,
+  name: string,
+  type?: 'airline' | 'bus' | 'train',
 };
 
-const styles = StyleSheet.create({
-  listItem: {
-    fontSize: 12,
-    padding: 2,
-    margin: 2,
-    backgroundColor: defaultTokens.backgroundBody,
-  },
-});
+export type TripSectorProps = {
+  +id: string,
+  +arrival: string,
+  +arrivalTime: string,
+  +carrier: CarrierData,
+  +tripDate: string,
+  +departure: string,
+  +departureTime: string,
+  +duration: string,
+};
+
+export type Route = $PropertyType<ResultsListItemType, 'route'>;
+export type RouteItem = $ElementType<$NonMaybeType<Route>, number>; // number because arrays are number-indexed
+
+const dateFormat = 'ccc d LLL';
+
+class ResultListItem extends React.Component<Props> {
+  getDateTimeInterval = (
+    to: DateTime,
+    from: DateTime,
+    unit: string | string[] = ['hours', 'minutes'],
+  ) => {
+    return to.diff(from, unit).toObject();
+  };
+
+  intervalToString = (interval: Duration, type: 'days' | 'hours' = 'hours') => {
+    if (!interval) {
+      return null;
+    }
+    if (type === 'days') {
+      switch (interval.days) {
+        case 0:
+          return `${Math.floor(interval.hours)} hours`;
+        case 1:
+          return `${interval.days} day`;
+        default:
+          return `${interval.days} days`;
+      }
+    }
+    return (
+      (interval.hours ? interval.hours + 'h' : '') +
+      (interval.minutes ? ' ' + interval.minutes + 'm' : '')
+    );
+  };
+
+  // @TODO Handle cases when we don't have some data from GraphQL/REST server
+  getSanitizedRouteItem = (route: RouteItem): ?TripSectorProps => {
+    if (route) {
+      const localArrival = DateTime.fromISO(route.arrival?.localTime);
+      const localDeparture = DateTime.fromISO(route.departure?.localTime);
+      const utcArrival = DateTime.fromISO(route.arrival?.utcTime);
+      const utcDeparture = DateTime.fromISO(route.departure?.utcTime);
+      return {
+        arrival: route.arrival?.city ?? '',
+        arrivalTime:
+          (localArrival && localArrival.toLocaleString(DateTime.TIME_SIMPLE)) ??
+          '',
+        carrier: { code: route.airline ?? '', name: '' }, // @TODO: get name of the airline
```

Comment:
Are there missing data for airline name in API or ?

Reply:
Exactly